### PR TITLE
dcp-680 Export link updates

### PR DESCRIPTION
**Pull request by aaclan-ebi**
dcp-680 
ebi-ait/dcp-ingest-central#680

### CODE REVIEW

File: src/main/java/org/humancellatlas/ingest/exporter/DefaultExporter.java [hunk 1]
```diff
@@ -103,27 +102,31 @@ public void exportProcesses(SubmissionEnvelope envelope) {
 
         ExportJob exportJob = exportJobService.createExportJob(envelope, exportJobRequest);
 
+        updateDcpVersionAndSendMessageForEachProcess(assayingProcessIds, exportJob);
+
+    }
+
+    private void updateDcpVersionAndSendMessageForEachProcess(Collection<String> assayingProcessIds, ExportJob exportJob) {
+        int totalCount = assayingProcessIds.size();
+        IndexCounter counter = new IndexCounter();
```
Comment (amnonkhen) [inline src/main/java/org/humancellatlas/ingest/exporter/DefaultExporter.java:111]: This counter could also hold the `totalCount` and generate the new `ExperimentProcess`es, can't it?

File: src/main/java/org/humancellatlas/ingest/export/job/ExportJobRepository.java [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package org.humancellatlas.ingest.export.job;
 
+import org.humancellatlas.ingest.export.ExportState;
```
[the remark this answers is not in the document]
Comment (aaclan-ebi) [reply src/main/java/org/humancellatlas/ingest/export/job/ExportJobRepository.java:3]: yeah, i reverted some changes, and forgot to remove imports, removing them

File: src/main/java/org/humancellatlas/ingest/export/job/ExportJobService.java [hunk 3]
```diff
@@ -1,6 +1,5 @@
 package org.humancellatlas.ingest.export.job;
 
-import java.util.UUID;
```
[the remark this answers is not in the document]
Comment (aaclan-ebi) [reply src/main/java/org/humancellatlas/ingest/export/job/ExportJobService.java:3]: yes

File: src/main/java/org/humancellatlas/ingest/exporter/DefaultExporter.java [hunk 4]
```diff
@@ -75,9 +76,8 @@ public void exportManifests(SubmissionEnvelope envelope) {
         int partitionSize = 500;
         partitionProcessIds(assayingProcessIds, partitionSize)
                 .stream()
-                .map(processIdBatch -> processService.getProcesses(processIdBatch))
-                .flatMap(Function.identity())
-                .map(process -> new ExperimentProcess(counter.next(), totalCount, process))
+                .flatMap(processIdBatch -> processService.getProcesses(processIdBatch))
+                .map(process -> new ExperimentProcess(counter.next(), totalCount, process, process.getSubmissionEnvelope(), process.getProject()))
```
[the remark this answers is not in the document]
Comment (aaclan-ebi) [reply src/main/java/org/humancellatlas/ingest/exporter/DefaultExporter.java]: I created a static factory method for ExperimentProcess, pls check.

File: src/main/java/org/humancellatlas/ingest/exporter/DefaultExporter.java (hunk 1 above)
Comment (aaclan-ebi) [reply src/main/java/org/humancellatlas/ingest/exporter/DefaultExporter.java]: Applied with your suggestion on using a factory method.

**Review by jacobwindsor — APPROVED** [review]

**Review by ke4 — APPROVED** [review]
LGTM

**Pull request merged**